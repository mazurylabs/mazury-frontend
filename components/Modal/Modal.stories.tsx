import { ComponentMeta } from '@storybook/react';
import { Button, Input } from 'components';
import React, { useState } from 'react';

import { Modal } from './Modal';

export default {
  title: 'Components/Modal',
  component: Modal,
  argTypes: {},
} as ComponentMeta<typeof Modal>;

export const Default = () => {
  const [isOpen, setIsOpen] = useState(false);

  return (
    <div>
      <Button onClick={() => setIsOpen(true)}>Open Modal</Button>
      <Modal isOpen={isOpen} onClose={() => setIsOpen(false)}>
        <h3 className="font-demi text-3xl text-indigoGray-90">
          Verify Twitter link
        </h3>
        <span className="mt-2 text-indigoGray-60">
          Paste your tweet&apos;s link here to verify
        </span>

        <Input
          label="Twitter link"
          outerClassName="mt-4"
          placeholder="e.g. https://twitter.com/mazuryxyz"
        />

        <div className="mt-4 flex w-full justify-around gap-4">
          <Button variant="secondary">SKIP</Button>
          <Button variant="primary">RETRY</Button>
        </div>
      </Modal>
    </div>
  );
};
