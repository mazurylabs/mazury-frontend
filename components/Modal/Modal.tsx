import { FC, useEffect } from 'react';
import ReactModal from 'react-modal';

interface ModalProps {
  children: React.ReactNode;
  isOpen: boolean;
  onClose: () => void;
  containerClassName?: string;
  appRootId?: string;
}

export const Modal: FC<ModalProps> = ({
  children,
  isOpen,
  onClose,
  containerClassName,
  appRootId = '__next',
}) => {
  useEffect(() => {
    ReactModal.setAppElement(document.getElementById(appRootId)!);
  }, [appRootId]);

  return (
    <ReactModal
      isOpen={isOpen}
      onRequestClose={onClose}
      shouldCloseOnEsc
      shouldCloseOnOverlayClick
      className={`absolute top-[50%] left-[50%] translate-x-[-50%] translate-y-[-50%] rounded-xl bg-white bg-opacity-100 py-4 px-6 shadow-xl ${containerClassName}`}
      data-testid="modal-container"
      overlayClassName="bg-black-900 bg-opacity-50 h-screen w-screen absolute top-0 left-0"
    >
      {children}
    </ReactModal>
  );
};
