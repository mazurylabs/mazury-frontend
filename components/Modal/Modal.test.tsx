import { fireEvent, render, screen } from '@testing-library/react';
import { useState } from 'react';
import { Modal } from './Modal';

const ModalContainer = () => {
  const [isOpen, setIsOpen] = useState(false);

  return (
    <div id="root-el">
      <button onClick={() => setIsOpen(true)}>Open Modal</button>
      <Modal
        isOpen={isOpen}
        onClose={() => setIsOpen(false)}
        appRootId="root-el"
      >
        <h1>Modal</h1>
      </Modal>
    </div>
  );
};

describe('Modal', () => {
  it('renders correctly and close functionality works properly', () => {
    render(<ModalContainer />);
    const toggleButton = screen.getByRole('button', { name: 'Open Modal' });
    fireEvent.click(toggleButton);
    const content = screen.getByText('Modal');
    const container = document.getElementsByClassName('ReactModalPortal')[0];
    expect(content).toBeInTheDocument();
    expect(container).toBeInTheDocument();
    expect(container).toMatchSnapshot();
    const overlay = document.getElementsByClassName('ReactModal__Overlay')[0];
    fireEvent.click(overlay);
    expect(content).not.toBeInTheDocument();
  });
});
