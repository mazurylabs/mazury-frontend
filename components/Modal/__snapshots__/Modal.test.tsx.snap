// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal renders correctly and close functionality works properly 1`] = `
<div
  class="ReactModalPortal"
>
  <div
    class="ReactModal__Overlay bg-black-900 bg-opacity-50 h-screen w-screen absolute top-0 left-0 z-20"
  >
    <div
      aria-modal="true"
      class="ReactModal__Content absolute top-[50%] left-[50%] translate-x-[-50%] translate-y-[-50%] rounded-xl bg-white bg-opacity-100 py-4 px-6 shadow-xl outline-none undefined"
      role="dialog"
      tabindex="-1"
    >
      <h1>
        Modal
      </h1>
    </div>
  </div>
</div>
`;
